packages/DevTools/src/signature-editor/function-signature-editor.ts

```diff
@@ -3,285 +3,88 @@ import * as ui from 'datagrok-api/ui';
-  'Type': FUNC_PARAM_FIELDS.TYPE,
-  'Name': FUNC_PARAM_FIELDS.NAME,
-  'Default value': FUNC_PARAM_FIELDS.DEFAULT_VALUE,
-  'Description': FUNC_PARAM_FIELDS.DESCRIPTION,
-  'Category': FUNC_PARAM_FIELDS.CATEGORY,
-};
-
-enum COMMON_TAG_NAME {
-  VALIDATORS = 'validators',
-  CAPTION = 'caption',
-  POSTFIX = 'postfix',
-  UNITS = 'units',
-  EDITOR = 'editor',
-  SEM_TYPE = 'semType',
-}
-
-enum OPTIONAL_TAG_NAME {
-  COLUMNS = 'columns',
-  TYPE = 'type',
-  FORMAT = 'format',
-  ALLOW_NULLS = 'allowNulls',
-  ACTION = 'action',
-  CHOICES = 'choices',
-  SUGGESTIONS = 'suggestions',
-  MIN = 'min',
-  MAX = 'max',
-}
-
-const COMMON_TAG_NAMES = [...Object.values(COMMON_TAG_NAME)];
-const DF_TAG_NAMES = [
-  ...Object.values(COMMON_TAG_NAME),
-  OPTIONAL_TAG_NAME.COLUMNS,
-  OPTIONAL_TAG_NAME.ACTION];
-const COLUMN_TAG_NAMES = [...Object.values(COMMON_TAG_NAME), OPTIONAL_TAG_NAME.TYPE, OPTIONAL_TAG_NAME.FORMAT,
-  OPTIONAL_TAG_NAME.ALLOW_NULLS, OPTIONAL_TAG_NAME.ACTION];
-const STRING_TAG_NAMES = [...Object.values(COMMON_TAG_NAME),
-  OPTIONAL_TAG_NAME.CHOICES , OPTIONAL_TAG_NAME.SUGGESTIONS];
-const INT_TAG_NAMES = [...Object.values(COMMON_TAG_NAME), OPTIONAL_TAG_NAME.MIN, OPTIONAL_TAG_NAME.MAX];
-
-const optionTags = ((param: DG.Property) => {
-  switch (param.propertyType) {
-    case DG.TYPE.INT:
-      return INT_TAG_NAMES;
-    case DG.TYPE.DATA_FRAME:
-      return DF_TAG_NAMES;
-    case DG.TYPE.COLUMN_LIST:
-    case DG.TYPE.COLUMN:
-      return COLUMN_TAG_NAMES;
-    case DG.TYPE.STRING:
-      return STRING_TAG_NAMES;
-    default:
-      return COMMON_TAG_NAMES;
-  }
-});
-
-enum LANGUAGE {
-  JS = 'javascript',
-  PYTHON = 'python',
-  R = 'r',
-  JULIA = 'julia',
-  OCTAVE = 'octave',
-  NODEJS = 'nodejs',
-  GROK = 'grok',
-  SQL = 'sql'
-}
-const languages = ['javascript', 'python', 'r', 'julia', 'octave', 'nodejs', 'grok', 'sql'];
-
-const headerSign = (lang: LANGUAGE) => {
-  switch (lang) {
-    case LANGUAGE.JS:
-    case LANGUAGE.NODEJS:
-      return '//';
-    case LANGUAGE.R:
-    case LANGUAGE.GROK:
-    case LANGUAGE.JULIA:
-    case LANGUAGE.PYTHON:
-    case LANGUAGE.OCTAVE:
-      return '#';
-    case LANGUAGE.SQL:
-      return '--';
-  }
-};
-
-let CONNECTION;
-
-async function getEditorSql(sql: string): Promise<DG.DataQuery> {
+async function getDataQuery(sql: string): Promise<DG.DataQuery> {
   const regex = /--connection:\s*(\S+)/;
   const match = sql.match(regex);
-  let connectionName;
-  if (match) 
-    connectionName = match[1];
+  const connectionName = match ? match[1]: null;
   const connection = await grok.functions.eval(connectionName);
-  CONNECTION = connectionName;
+  conn = connectionName;
   const query = connection.query('sql', sql);
   return query;
 }
 
+async function getInputCode(v: DG.View, functionCode: string) {
+  let inputCodeCopy;
+  v.type === DATA_QUERY_VIEW 
+  ? inputCodeCopy = await getDataQuery(functionCode) 
+  : inputCodeCopy = DG.Script.create(functionCode);
```

Comment:
you don't need to create null variable `inputCodeCopy`, you can assign it to a ternary:
```
const inputCodeCopy = v.type === DATA_QUERY_VIEW ?
  await getDataQuery(functionCode) :
  DG.Script.create(functionCode);
```